samples/star/src/main/kotlin/com/slack/circuit/star/repo/PetRepository.kt

```diff
@@ -2,49 +2,176 @@
 // SPDX-License-Identifier: Apache-2.0
 package com.slack.circuit.star.repo
 
-import com.slack.circuit.star.data.Animal
+import android.content.Context
+import app.cash.sqldelight.EnumColumnAdapter
+import app.cash.sqldelight.coroutines.asFlow
+import app.cash.sqldelight.coroutines.mapToList
+import app.cash.sqldelight.driver.android.AndroidSqliteDriver
 import com.slack.circuit.star.data.PetfinderApi
+import com.slack.circuit.star.db.Animal as DbAnimal
+import com.slack.circuit.star.db.AnimalBio
+import com.slack.circuit.star.db.Gender
+import com.slack.circuit.star.db.ImmutableListAdapter
+import com.slack.circuit.star.db.OpJournal
+import com.slack.circuit.star.db.Size
+import com.slack.circuit.star.db.StarDatabase
 import com.slack.circuit.star.di.AppScope
+import com.slack.circuit.star.di.ApplicationContext
 import com.slack.circuit.star.di.SingleIn
 import com.squareup.anvil.annotations.ContributesBinding
+import java.time.Duration
+import java.time.Instant
+import java.time.ZoneOffset
+import java.util.Locale
 import javax.inject.Inject
+import kotlin.LazyThreadSafetyMode.NONE
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import retrofit2.HttpException
 
 interface PetRepository {
-  suspend fun getAnimals(forceRefresh: Boolean): List<Animal>
-  suspend fun getAnimal(id: Long): Animal?
+  suspend fun refreshData()
+  fun animalsFlow(): Flow<List<DbAnimal>>
+  suspend fun getAnimal(id: Long): DbAnimal?
   suspend fun getAnimalBio(id: Long): String?
 }
 
 @SingleIn(AppScope::class)
 @ContributesBinding(AppScope::class)
-class PetRepositoryImpl @Inject constructor(private val petFinderApi: PetfinderApi) :
-  PetRepository {
-  private lateinit var animals: List<Animal>
+class PetRepositoryImpl
+@Inject
+constructor(
+  @ApplicationContext private val appContext: Context,
+  private val petFinderApi: PetfinderApi,
+) : PetRepository {
 
-  override suspend fun getAnimals(forceRefresh: Boolean): List<Animal> {
-    if (!this::animals.isInitialized || forceRefresh) fetchAnimals()
-    return animals
+  private val backgroundScope = CoroutineScope(Job() + IO)
```

Comment:
Should is `SupervisorJob()` used.
A failed coroutine should not cancel the remaining siblings 😁